### PR TITLE
rep migration

### PR DESCRIPTION
**Pull request by nuevoalex**
[Clubhouse Story](https://app.clubhouse.io/augur/story/7282)
[Clubhouse Story](https://app.clubhouse.io/augur/story/7286)

## Submitter checklist
- [X] Test covering functionality added/fixed in
- Check the linked story includes the following:
  - [X] Steps to reproduce
  - [ ] Screenshot (If applicable)

## Reviewer Checklist
- [ ] Test/lint pass 
- [ ] Post merge, story marked complete

### CODE REVIEW

File: src/modules/forking/components/forking-notification/forking-notification.jsx [hunk 1]
```diff
@@ -16,6 +16,12 @@ class ForkingNotification extends Component {
     this.expand = this.expand.bind(this)
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.location !== this.props.location) {
+      this.state.isExpanded = false
```
Comment (justinbarry) [inline src/modules/forking/components/forking-notification/forking-notification.jsx]: We should use this.setState here.

File: src/modules/forking/components/migrate-rep-form/migrate-rep-form.jsx [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    if (this.props.repAmount) this.state.inputRepAmount = this.props.repAmount.toString()
+
+    this.componentWillReceiveProps(this.props)
+
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.selectedOutcome || typeof nextProps.selectedOutcome === 'number') {
+      if (!this.state.outcomes.find(o => o.id === nextProps.selectedOutcome)) {
+        this.state.inputSelectedOutcome = nextProps.selectedOutcome
```
Comment (justinbarry) [inline src/modules/forking/components/migrate-rep-form/migrate-rep-form.jsx]: Use this.setState here. See: https://reactjs.org/docs/react-component.html#componentwillreceiveprops

Comment (justinbarry) [reply src/modules/forking/components/migrate-rep-form/migrate-rep-form.jsx]: ....and given you are calling this from the constructor, react may not be happy.

File: src/modules/forking/components/migrate-rep-form/migrate-rep-form.jsx [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+    delete updatedValidations.err
+    let isInvalid = isMarketInValid
+
+    // outcome with id of .5 means invalid
+    if (selectedOutcome === '0.5') isInvalid = true
+
+    MigrateRepForm.checkRepAmount(this.props.repAmount, updatedValidations)
+
+    this.state.inputSelectedOutcome = ''
+    this.state.maxRep = this.calculateMaxRep(selectedOutcome)
```
Comment (justinbarry) [inline src/modules/forking/components/migrate-rep-form/migrate-rep-form.jsx]: Probably should use setState here as well.

**Comment by nuevoalex** [comment]
Should be good now!

**Pull request merged**